### PR TITLE
Fix multi cluster management with metadata syncing enabled

### PR DESCRIPTION
**Pull request by halilozanakgul**
Fixes multi_cluster_management tests fails with MX

### CODE REVIEW

**Comment by onderkalaci** [comment]
can you let me know once this is ready for review after merging the other PRs and rebasing this one?

File: src/test/regress/expected/multi_cluster_management.out [hunk 1]
```diff
@@ -223,6 +223,22 @@ SELECT 1 FROM master_add_secondary_node('localhost', :worker_2_port + 2, 'localh
 ERROR:  permission denied for function master_add_secondary_node
 SELECT master_update_node(nodeid, 'localhost', :worker_2_port + 3) FROM pg_dist_node WHERE nodeport = :worker_2_port;
 ERROR:  permission denied for function master_update_node
+RESET ROLE;
+ALTER SYSTEM SET citus.enable_metadata_sync_by_default TO OFF;
```
[the remark this answers is not in the document]
Comment (halilozanakgul) [reply src/test/regress/expected/multi_cluster_management.out]: Removed this after #5402

**Comment by onderkalaci** [comment]
> DESCRIPTION: Fixes multi_cluster_management tests fails with MX

I think we don't need DESCRIPTION. These are only useful for auto-creating the release notes, and this change does not seem to be a good candidate for release notes

File: src/test/regress/expected/multi_cluster_management.out [hunk 2]
```diff
@@ -882,6 +1049,15 @@ SELECT * FROM pg_dist_node WHERE nodeid = :worker_1_node;
...
 
+SET client_min_messages TO ERROR;
+SELECT start_metadata_sync_to_node(nodename, nodeport) FROM pg_dist_node WHERE isactive = 't' and noderole = 'primary';
+ start_metadata_sync_to_node
+---------------------------------------------------------------------
+
+
+(2 rows)
+
+RESET client_min_messages;
```
Comment (onderkalaci) [inline src/test/regress/expected/multi_cluster_management.out:1060]: I think at the end of this file, we should have a test which verifies that all the primary nodes have metadata synced:
```SQL
-- verify that at the end of this file, all primary nodes have metadata synced
SELECT bool_and(hasmetadata) & bool_and(metadatasynced) FROM pg_dist_node WHERE isactive = 't' and noderole = 'primary';
```

This file is way too complex, and we should guard for future changes.

File: src/test/regress/expected/turn_mx_on_0.out [hunk 3]
```diff
@@ -0,0 +1,16 @@
+ALTER SYSTEM SET citus.enable_metadata_sync_by_default TO ON;
```
Comment (onderkalaci) [inline src/test/regress/expected/turn_mx_on_0.out:1]: I want to get rid of these as quickly as possible :)

**Review by onderkalaci — APPROVED** [review]
Lets make sure that the tests in enterprise passes before merging this PR

File: src/test/regress/expected/multi_cluster_management.out [hunk 4]
```diff
@@ -309,6 +387,13 @@ HINT:  To proceed, either drop the distributed tables or use undistribute_table(
 -- mark all placements in the candidate node as inactive
 SELECT groupid AS worker_2_group FROM pg_dist_node WHERE nodeport=:worker_2_port \gset
 UPDATE pg_dist_placement SET shardstate=3 WHERE groupid=:worker_2_group;
+SELECT run_command_on_workers('UPDATE pg_dist_placement SET shardstate=3 WHERE groupid=' || :'worker_2_group');
```
[the remark this answers is not in the document]
Comment (halilozanakgul) [reply src/test/regress/expected/multi_cluster_management.out:390]: Should I do that after this PR?

[the remark this answers is not in the document]
Comment (onderkalaci) [reply src/test/regress/expected/multi_cluster_management.out:390]: that works too

File: src/test/regress/expected/multi_cluster_management.out [hunk 5]
```diff
@@ -184,27 +184,96 @@ NOTICE:  Replicating reference table "test_reference_table" to the node localhos
...
+(2 rows)
+
 GRANT EXECUTE ON FUNCTION master_activate_node(text,int) TO node_metadata_user;
 GRANT EXECUTE ON FUNCTION master_add_inactive_node(text,int,int,noderole,name) TO node_metadata_user;
 GRANT EXECUTE ON FUNCTION master_add_node(text,int,int,noderole,name) TO node_metadata_user;
 GRANT EXECUTE ON FUNCTION master_add_secondary_node(text,int,text,int,name) TO node_metadata_user;
 GRANT EXECUTE ON FUNCTION master_disable_node(text,int) TO node_metadata_user;
 GRANT EXECUTE ON FUNCTION master_remove_node(text,int) TO node_metadata_user;
 GRANT EXECUTE ON FUNCTION master_update_node(int,text,int,bool,int) TO node_metadata_user;
+SELECT run_command_on_workers('GRANT EXECUTE ON FUNCTION master_activate_node(text,int) TO node_metadata_user');
```
[the remark this answers is not in the document]
Comment (halilozanakgul) [reply src/test/regress/expected/multi_cluster_management.out]: I couldn't.
While trying to explain why we need them, I noticed we don't.

File: src/test/regress/expected/turn_mx_on_0.out (hunk 3 above)
Comment (halilozanakgul) [reply src/test/regress/expected/turn_mx_on_0.out:1]: Me too

**Pull request merged**